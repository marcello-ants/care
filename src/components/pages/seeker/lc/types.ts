import { GeneralProviderProfile } from '@/types/seeker';

export interface ProviderProfile extends GeneralProviderProfile {
  numberOfReviews: number;
  qualities?: ProviderQualities;
  services?: ProviderServices;
  firstName: string;
}

export interface ProviderQualities {
  certifiedNursingAssistant?: boolean;
  certifiedRegisterNurse?: boolean;
  certifiedTeacher?: boolean;
  childDevelopmentAssociate?: boolean;
  comfortableWithPets?: boolean;
  cprTrained?: boolean;
  crn?: boolean;
  doesNotSmoke?: boolean;
  doula?: boolean;
  earlyChildDevelopmentCoursework?: boolean;
  earlyChildhoodEducation?: boolean;
  firstAidTraining?: boolean;
  nafccCertified?: boolean;
  ownTransportation?: boolean;
  specialNeedsCare?: boolean;
  trustlineCertifiedCalifornia?: boolean;
}

export interface ProviderServices {
  carpooling?: boolean;
  craftAssistance?: boolean;
  errands?: boolean;
  groceryShopping?: boolean;
  laundryAssistance?: boolean;
  lightHousekeeping?: boolean;
  mealPreparation?: boolean;
  swimmingSupervision?: boolean;
  travel?: boolean;
}
