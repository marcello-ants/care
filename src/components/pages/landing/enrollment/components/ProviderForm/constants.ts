export const PROVIDER_SERVICES_DESKTOP = [
  {
    description: 'Child care',
    label: 'Child care',
    value: 'CHILD_CARE',
  },
  {
    description: 'Senior care',
    label: 'Senior care',
    value: 'SENIOR_CARE',
  },
  {
    description: 'Housekeeping',
    label: 'Housekeeping',
    value: 'HOUSEKEEPING',
  },
  {
    description: 'Pet care',
    label: 'Pet care',
    value: 'PET_CARE',
  },
  {
    description: 'Tutoring',
    label: 'Tutoring',
    value: 'TUTORING',
  },
  {
    description: 'Special needs care',
    label: 'Special needs care',
    value: 'SPECIAL_NEEDS',
  },
  {
    description: 'Errands & odd jobs',
    label: 'Errands & odd jobs',
    value: 'GENERIC',
  },
];

export const PROVIDER_SERVICES_MOBILE = [
  {
    description: 'Child care',
    label: 'Child care',
    value: 'CHILD_CARE',
  },
  {
    description: 'Senior care',
    label: 'Senior care',
    value: 'SENIOR_CARE',
  },
  {
    description: 'Pet care',
    label: 'Pet care',
    value: 'PET_CARE',
  },
  {
    description: 'Tutoring',
    label: 'Tutoring',
    value: 'TUTORING',
  },
  {
    description: 'Special needs care',
    label: 'Special needs care',
    value: 'SPECIAL_NEEDS',
  },
  {
    description: 'Housekeeping',
    label: 'Housekeeping',
    value: 'HOUSEKEEPING',
  },
  {
    description: 'Errands & odd jobs',
    label: 'Errands & odd jobs',
    value: 'GENERIC',
  },
];

export const DROPOFF_URLS = {
  CHILD_CARE: '/swf/ccOptimizedFlow',
  SENIOR_CARE: '/app/enrollment/provider/sc ',
  PET_CARE: '/swf/pcOptimizedFlow',
  HOUSEKEEPING: '/swf/hkOptimizedFlow',
  TUTORING: '/swf/tuOptimizedFlow',
  SPECIAL_NEEDS: '/swf/snOptimizedFlow',
  GENERIC: '/swf/cgOptimizedFlow',
  AU_PAIR: '',
  CARE_GIGS: '',
};
