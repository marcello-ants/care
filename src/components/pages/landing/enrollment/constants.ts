export const SITTER_FORM = {
  providerServices: {
    title: 'What Services do you offer?',
    id: 'providerService',
    options: [
      {
        value: 'CHILD_CARE',
        text: 'Babysitter',
      },
      {
        value: 'CHILD_CARE',
        text: 'Nanny',
      },
      {
        value: 'SPECIAL_NEEDS',
        text: 'Special Needs Provider',
      },
      {
        value: 'TUTORING',
        text: 'Tutor',
      },
      {
        value: 'TUTORING',
        text: 'Private Lesson Instructor',
      },
      {
        value: 'SENIOR_CARE',
        text: 'Senior Care Provider',
      },
      {
        value: 'PET_CARE',
        text: 'Pet Care Provider',
      },
      {
        value: 'HOUSEKEEPING',
        text: 'Housekeeper',
      },
      {
        value: 'GENERIC',
        text: 'Errands &amp; Odd Jobs Provider',
      },
    ],
  },
  providerEmail: {
    id: 'email',
    placeholder: 'Email',
  },
  providerPassword: {
    id: 'password',
    placeholder: 'Password',
  },
  providerZip: {
    id: 'providerZip',
    placeholder: 'ZIP',
  },
};

export const GENERIC_FORM = {
  howDidYouHearAboutUs: {
    title: 'How did you hear about us?',
    id: 'howDidYouHearAboutUs',
    options: [
      {
        value: 'FACEBOOK',
        text: 'Facebook',
      },
      {
        value: 'BANNER_AD',
        text: 'Banner Ad',
      },
      {
        value: 'TV_AD',
        text: 'TV Ad',
      },
      {
        value: 'OTHER_SOCIAL_MEDIA',
        text: 'Other Social Media (Twitter, Instagram, Pinterest, LinkedIn)',
      },
      {
        value: 'PRESS_COVERAGE',
        text: 'Press Coverage (News, Magazine, Blog)',
      },
      {
        value: 'FRIENDS_OR_FAMILY',
        text: 'Friends or Family',
      },
      {
        value: 'ONLINE_VIDEO',
        text: 'Online Video (Hulu, Youtube)',
      },
      {
        value: 'SEARCH_ENGINE',
        text: 'Search Engine (Google, Bing)',
      },
      {
        value: 'PARENTING_GROUP',
        text: 'Parenting Group or Forum',
      },
      {
        value: 'OTHER',
        text: 'Other',
      },
    ],
  },
};
