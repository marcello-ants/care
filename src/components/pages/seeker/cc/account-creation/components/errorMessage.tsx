import React from 'react';
import { Box, Grid } from '@material-ui/core';
import { Alert } from '@material-ui/lab';

type ErrorMessageData = {
  errorMessageText: string;
};

const ErrorMessageNotification = (props: ErrorMessageData) => {
  const { errorMessageText } = props;

  return (
    <Grid item xs={12}>
      <Box mb={3}>
        <Alert severity="error">{errorMessageText}</Alert>
      </Box>
    </Grid>
  );
};

export default ErrorMessageNotification;
