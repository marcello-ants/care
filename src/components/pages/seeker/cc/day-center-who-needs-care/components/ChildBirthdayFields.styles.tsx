// External Dependencies
import { makeStyles } from '@material-ui/core';
import { theme } from '@care/material-ui-theme';

const useStyles = makeStyles({
  childContainer: {
    padding: theme.spacing(0),
  },
  childContainerNth: {
    marginTop: theme.spacing(4),
    position: 'relative',
  },
  optionsContainer: {
    display: 'flex',
    flexDirection: 'row',
  },
  fullWidth: {
    width: '100%',
  },
  error: {
    color: theme.palette.care.red[800],
    paddingLeft: theme.spacing(1),
  },
  selectContainer: {
    marginTop: theme.spacing(1),
  },
  monthSelect: {
    marginRight: theme.spacing(0.5),
    '& label': {
      paddingLeft: theme.spacing(1),
    },
  },
  yearSelect: {
    marginLeft: theme.spacing(0.5),
    '& label': {
      paddingLeft: theme.spacing(1),
    },
  },
  removeChildButton: {
    position: 'absolute',
    top: '0',
    right: '0',
    width: 'auto',
    height: 'auto',
    lineHeight: 'unset',
    minWidth: 'auto',
    padding: theme.spacing(0),
    color: theme.palette.care.blue[700],
    '&:hover': {
      backgroundColor: 'transparent',
    },
  },
});

export default useStyles;
