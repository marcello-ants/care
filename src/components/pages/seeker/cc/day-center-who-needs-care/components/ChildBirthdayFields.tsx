// External Dependencies
import React, { useMemo } from 'react';
import { Button, FormHelperText, Grid, InputLabel, MenuItem } from '@material-ui/core';
import { Select, Typography } from '@care/react-component-lib';

// Internal Dependencies
import { ChildDateOfBirth } from '@/types/seekerCC';
import { getBirthAvailableMonthsAndYears } from '../utils';
import useStyles from './ChildBirthdayFields.styles';

// Interfaces
interface TErrorProps {
  year?: string;
  month?: string;
}

type SelectInputProps = React.ComponentProps<typeof Select>;
interface IChildBirthdayFieldsProps {
  child: ChildDateOfBirth;
  onRemoveChild: () => void;
  onChange: SelectInputProps['onChange'];
  childIndex: number;
  error?: TErrorProps;
  allow9MonthsOutFromNow?: boolean;
}

function ChildBirthdayFields({
  child,
  childIndex: i,
  onChange,
  onRemoveChild,
  error,
  allow9MonthsOutFromNow,
}: IChildBirthdayFieldsProps) {
  const classes = useStyles();

  const { months, years } = useMemo(
    () => getBirthAvailableMonthsAndYears(child.month, child.year, allow9MonthsOutFromNow),
    [child.year, child.month, allow9MonthsOutFromNow]
  );

  const monthsOptions = useMemo(() => {
    return months.map((month) => {
      return (
        <MenuItem key={month.value} value={month.value}>
          {month.name}
        </MenuItem>
      );
    });
  }, [months]);

  const yearsOptions = useMemo(() => {
    return years.map((year: string) => {
      return (
        <MenuItem key={year} value={year}>
          {year}
        </MenuItem>
      );
    });
  }, [years]);

  return (
    <Grid
      // eslint-disable-next-line react/no-array-index-key
      item
      xs={12}
      className={`${
        i === 0
          ? `${classes.childContainer}`
          : `${classes.childContainer} ${classes.childContainerNth}`
      }`}>
      <Typography variant="h3">Child {i + 1}</Typography>
      {i !== 0 && (
        <Button className={classes.removeChildButton} onClick={onRemoveChild}>
          Remove Child
        </Button>
      )}
      <Grid item xs={12} className={classes.optionsContainer}>
        <Grid item xs={6} className={`${classes.selectContainer} ${classes.monthSelect}`}>
          <InputLabel id={`month-select-label-${i}`}>Birth month</InputLabel>
          <Select
            className={classes.fullWidth}
            labelId={`month-select-label-${i}`}
            id={`month-select-child-${i}`}
            name={`children[${i}].month`}
            onChange={onChange}
            value={child.month}
            error={Boolean(error?.month)}>
            {monthsOptions}
          </Select>
          {error?.month && (
            <FormHelperText className={classes.error}>Select their birth month</FormHelperText>
          )}
        </Grid>
        <Grid item xs={6} className={`${classes.selectContainer} ${classes.yearSelect}`}>
          <InputLabel id={`year-select-label-${i}`}>Birth year</InputLabel>
          <Select
            className={classes.fullWidth}
            labelId={`year-select-label-${i}`}
            id={`year-select-child-${i}`}
            name={`children[${i}].year`}
            onChange={onChange}
            value={child.year}
            error={Boolean(error?.year)}>
            {yearsOptions}
          </Select>
          {error?.year && (
            <FormHelperText className={classes.error}>Select their birth year</FormHelperText>
          )}
        </Grid>
      </Grid>
    </Grid>
  );
}

ChildBirthdayFields.defaultProps = {
  error: undefined,
  allow9MonthsOutFromNow: undefined,
};

export type { IChildBirthdayFieldsProps, TErrorProps };
export default ChildBirthdayFields;
