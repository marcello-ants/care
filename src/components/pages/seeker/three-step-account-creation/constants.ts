import * as yup from 'yup';
import { ServiceType } from '@/__generated__/globalTypes';
import { VerticalsAbbreviation } from '@/constants';

export const validationSchema = yup.object({
  firstName: yup
    .string()
    .min(2, 'First name needs to have at least two letters.')
    .required('First name is required'),
  lastName: yup
    .string()
    .min(2, 'Last name needs to have at least two letters.')
    .required('Last name is required'),
});

export const SERVICE_TYPE_BY_VERTICAL: { [key: string]: ServiceType } = {
  HK: ServiceType.HOUSEKEEPING,
  PC: ServiceType.PET_CARE,
  TU: ServiceType.TUTORING,
  CC: ServiceType.CHILD_CARE,
  DC: ServiceType.CHILD_CARE,
  IB: ServiceType.CHILD_CARE,
  SC: ServiceType.SENIOR_CARE,
};

export const ERROR_CREATING_ACCOUNT = 'An error occurred creating your account, please try again.';

export const FACEBOOK_VERTICALS_ALLOWANCE = ['CC'];
export const HDYHAU_VERTICALS_ALLOWANCE: VerticalsAbbreviation[] = ['HK', 'TU', 'PC', 'DC'];
