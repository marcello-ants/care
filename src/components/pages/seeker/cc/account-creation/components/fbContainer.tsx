import { Box, makeStyles, Grid, Button } from '@material-ui/core';
import { useFacebookApiKey } from '@/components/FacebookApiKeyContext';

// @ts-ignore
import FacebookLogin from 'react-facebook-login/dist/facebook-login-render-props';
import { Icon48SocialMediaFacebook } from '@care/react-icons';

export type facebookResponseProps = {
  accessToken: string;
  // eslint-disable-next-line
  data_access_expiration_time?: number;
  email: string;
  expiresIn?: number;
  graphDomain?: string;
  id?: string;
  name: string;
  signedRequest?: string;
  userID?: string;
};

export interface SplitFormValues {
  email: string;
}

export interface SplitFormWithFbData extends SplitFormValues {
  firstName?: string;
  lastName?: string;
  accessToken?: string;
}

const mobileWidth = 360;
const fbColor = '#1877F2';
const fbIconSize = '35px';
const useStyles = (fbEmphasizedDesign: boolean) =>
  makeStyles((theme) => ({
    fbButton: {
      paddingTop: theme.spacing(5),
      borderColor: fbEmphasizedDesign ? fbColor : theme.palette.default?.dark,
      background: fbEmphasizedDesign ? fbColor : theme.palette.care?.white,
      color: fbEmphasizedDesign ? theme.palette.care?.white : theme.palette.default?.dark,
      fontSize: fbEmphasizedDesign ? '21px' : '18px',
      '&:hover': {
        background: fbEmphasizedDesign ? fbColor : theme.palette.care?.white,
      },
    },
    fbButtonContent: {
      display: 'grid',
      gridTemplateColumns: 'repeat(3, minmax(0, 1fr))',
      [theme.breakpoints.down(mobileWidth)]: {
        display: 'flex',
        alignItems: 'center',
      },
    },
    fbIconContainer: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      width: fbIconSize,
      '& svg': {
        width: fbIconSize,
        height: fbIconSize,
      },
    },
    fbIcon: {
      background: '#1a77f2',
      borderRadius: '19px',
    },
    fbButtonText: {
      justifySelf: 'center',
      marginLeft: '10px',
      marginRight: '5px',
      fontColor: theme.palette.care?.white,
    },
  }));

type GetFbContainerProps = {
  handleFacebookLogin: (response: facebookResponseProps) => Promise<void>;
  fbButtonOnTop: boolean;
  fbButtonOnBottom: boolean;
};

const getFbContainer = ({
  handleFacebookLogin,
  fbButtonOnTop,
  fbButtonOnBottom,
}: GetFbContainerProps) => {
  const fbEmphasizedDesign = fbButtonOnBottom || fbButtonOnTop;
  const classes = useStyles(fbEmphasizedDesign)();
  const { apiKey } = useFacebookApiKey();

  return (
    <Grid item xs={12}>
      <Box pt={fbButtonOnTop ? 0 : 2}>
        <FacebookLogin
          appId={apiKey}
          autoLoad={false}
          fields="name,email"
          callback={handleFacebookLogin}
          render={(renderProps: any) => (
            <Button
              onClick={renderProps.onClick}
              className={classes.fbButton}
              variant="outlined"
              size="large"
              fullWidth>
              <div className={classes.fbButtonContent}>
                <div className={classes.fbIconContainer}>
                  <Icon48SocialMediaFacebook size="38px" color="white" className={classes.fbIcon} />
                </div>
                <p className={classes.fbButtonText}>Continue with Facebook</p>
              </div>
            </Button>
          )}
        />
      </Box>
    </Grid>
  );
};

export default getFbContainer;
