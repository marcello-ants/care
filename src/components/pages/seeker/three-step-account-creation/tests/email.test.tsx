import { NextRouter } from 'next/router';
import { RouterContext } from 'next/dist/shared/lib/router-context';
import { render, waitFor, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import AuthService from '@/lib/AuthService';
import { AppStateProvider } from '@/components/AppState';
import { initialAppState } from '@/state';
import { AppState } from '@/types/app';
import useProviderCount from '@/components/hooks/useProviderCount';
import { FLOWS } from '@/constants';

import { ServiceType } from '@/__generated__/globalTypes';
import AccountCreationEmailForm from '../email';

let mockRouter: NextRouter;
let textboxEmail: HTMLElement;
let buttonNext: HTMLElement;

const TEST_DATA_ZIP_CODE = '02452';
const TEST_DATA_EMAIL = 'email@test.com';
const INVALID_TEST_DATA_EMAIL = 'email';

jest.mock('@/lib/AuthService');
jest.mock('@/components/hooks/useProviderCount');
const AuthServiceMock = AuthService as jest.Mock;
const redirectLoginSpy = jest.fn();
AuthServiceMock.mockImplementation(() => {
  return {
    redirectLogin: redirectLoginSpy,
  };
});

function renderPage(mocks: MockedResponse[] = []) {
  const initialState: AppState = {
    ...initialAppState,
    flow: {
      ...initialAppState.flow,
      flowName: FLOWS.SEEKER_HOUSEKEEPING.name,
    },
    seeker: {
      ...initialAppState.seeker,
      zipcode: TEST_DATA_ZIP_CODE,
    },
  };
  const pathname = '/seeker/hk/account-creation';
  // @ts-ignore
  mockRouter = {
    push: jest.fn(),
    asPath: pathname,
    pathname,
    basePath: '',
  };

  const view = render(
    <RouterContext.Provider value={mockRouter}>
      <MockedProvider mocks={mocks} addTypename={false}>
        <AppStateProvider initialStateOverride={initialState}>
          <AccountCreationEmailForm serviceType={ServiceType.HOUSEKEEPING} />
        </AppStateProvider>
      </MockedProvider>
    </RouterContext.Provider>
  );
  textboxEmail = screen.getByRole('textbox', { name: 'Email address' });
  buttonNext = screen.getByRole('button', { name: 'Next' });

  return view;
}

describe('Housekeeping - email account creation form', () => {
  const mockProviderCount = {
    displayProviderMessage: true,
    numOfProviders: 100,
  };
  const mockNoProviderCount = {
    displayProviderMessage: false,
    numOfProviders: 0,
  };

  (useProviderCount as jest.Mock).mockReturnValue(mockProviderCount);

  it('matches snapshots', () => {
    const { asFragment } = renderPage();

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly', () => {
    renderPage();
    expect(textboxEmail).toBeInTheDocument();
    expect(buttonNext).toBeInTheDocument();
  });

  it('renders housekeepers nearby banner', () => {
    renderPage();

    expect(screen.queryByTestId('housekeepers-nearby')).toHaveTextContent(
      'Nice! 100 housekeepers near you!'
    );
  });

  it("doesn't render housekeepers nearby banner if there are no housekeepers around", () => {
    (useProviderCount as jest.Mock).mockReturnValue(mockNoProviderCount);

    renderPage();

    expect(screen.queryByTestId('housekeepers-nearby')).not.toBeInTheDocument();
  });

  it('shows validation errors', async () => {
    renderPage();

    userEvent.click(buttonNext);
    expect(await screen.findByText(/Email is required/)).toBeInTheDocument();
  });
  it('invalid email error', async () => {
    renderPage();

    userEvent.type(textboxEmail, INVALID_TEST_DATA_EMAIL);
    fireEvent.blur(textboxEmail);
    await waitFor(() => {
      expect(textboxEmail).not.toBeValid();
    });

    userEvent.click(buttonNext);
    expect(await screen.findByText(/Please enter a valid email./)).toBeInTheDocument();
  });
  it('email successfully validated', async () => {
    renderPage();

    userEvent.type(textboxEmail, TEST_DATA_EMAIL);
    fireEvent.blur(textboxEmail);
    await waitFor(() => {
      expect(textboxEmail).toBeValid();
    });

    userEvent.click(buttonNext);
    expect(screen.queryByText(/Please enter a valid email./)).not.toBeInTheDocument();
  });
});
