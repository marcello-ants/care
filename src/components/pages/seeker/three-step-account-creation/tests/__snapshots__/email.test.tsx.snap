// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Housekeeping - email account creation form matches snapshots 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <div
      class="makeStyles-root-3"
    >
      <div
        class="makeStyles-content-4 makeStyles-information-8 makeStyles-roundCorners-6"
        style="width: 100%;"
      >
        <div
          class="makeStyles-icon-7"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-root-13"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M2 12a10 10 0 1110 10A10 10 0 012 12zm1.32 0a8.64 8.64 0 108.59-8.68A8.68 8.68 0 003.32 12z"
              fill="#182c42"
            />
            <path
              d="M12 17a.83.83 0 00.82-.83v-5.45a.83.83 0 10-1.65 0v5.45A.78.78 0 0012 17zM12 8.65a.83.83 0 10-.83-.82.82.82 0 00.83.82z"
              fill="#182c42"
            />
          </svg>
        </div>
        <div>
          <p
            class="MuiTypography-root MuiTypography-body2"
            data-testid="housekeepers-nearby"
          >
            Nice! 100 housekeepers near you!
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container"
  >
    <div
      class="MuiBox-root MuiBox-root-28"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2"
      >
        Create a free account
      </h2>
      <div
        class="MuiBox-root MuiBox-root-29"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          See housekeepers who match your needs. It only takes a few seconds.
        </p>
      </div>
      <div
        class="MuiGrid-root makeStyles-inputContainer-1 MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-inline-30"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
            for="email"
            id="email-label"
          >
            Email address
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              id="email"
              name="email"
              type="email"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiBox-root MuiBox-root-34"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Next
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
